### PR TITLE
BUGFIX: pdns-1283 pdns_server and pdns_recursor do not get along

### PR DESCRIPTION
**Pull request by UX-admin**
https://github.com/PowerDNS/pdns/issues/1283

### CODE REVIEW

File: dnsproxy.cc.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- pdns/dnsproxy.cc.orig       Thu Oct 30 06:18:22 2014
++++ pdns/dnsproxy.cc    Mon Dec  8 14:55:41 2014
+@@ -169,7 +169,7 @@
+       memcpy(&d,buffer,sizeof(d));
+       {
+         Lock l(&d_lock);
+-#if BYTE_ORDER == BIG_ENDIAN
++#ifdef _LITTLE_ENDIAN
```
Comment (pieterlexis) [inline dnsproxy.cc.patch]: Hi,

Where does the `_LITTLE_ENDIAN` definition come from, system sources? Why not use `#if BYTE_ORDER == LITTLE_ENDIAN` as defined (and used) in `dns.hh`?

Comment (zeha) [inline dnsproxy.cc.patch]: doesn't the change imply that the `#if` should go away completely?

Comment (UX-admin) [inline dnsproxy.cc.patch]: _LITTLE_ENDIAN comes from /usr/include/sys/isa_defs.h.

If the clause is completely removed, then the i86pc version does not work correctly, i86pc platform being little endian.

**Comment by Habbie** [comment]
Please reopen this as a pull request, and explain what it's for. Thank you!

**Pull request closed without merging**